pandas/core/generic.py

```diff
@@ -3620,6 +3619,8 @@ def astype(self, dtype, copy=True, errors='raise', **kwargs):
             the same type. Alternatively, use {col: dtype, ...}, where col is a
             column label and dtype is a numpy.dtype or Python type to cast one
             or more of the DataFrame's columns to column-specific types.
+        copy : bool, default True.
+            Return a copy when ``copy=True`` (be really careful with this!).
```

Comment:
while we are at this, I would like to see this explanation improved ("be really careful with this!" is not that helpful ..). 
But also ok to leave that for other issue/PR if you don't feel comfortable with it

(I don't know the internals well enough to know exactly how the keyword is used, but for numerical dtypes I suppose it follows the numpy `astype` behaviour? Which means only not copying if dtype (and order) is equivalent)